src/line.h

```diff
@@ -6,58 +6,72 @@
 #include "enums.h"
 #include <math.h>
 
-enum direction {
-    NORTH = 0,
-    NORTHEAST,
-    EAST,
-    SOUTHEAST,
-    SOUTH,
-    SOUTHWEST,
-    WEST,
-    NORTHWEST,
-    ABOVENORTH,
-    ABOVENORTHEAST,
-    ABOVEEAST,
-    ABOVESOUTHEAST,
-    ABOVESOUTH,
-    ABOVESOUTHWEST,
-    ABOVEWEST,
-    ABOVENORTHWEST,
-    BELOWNORTH,
-    BELOWNORTHEAST,
-    BELOWEAST,
-    BELOWSOUTHEAST,
-    BELOWSOUTH,
-    BELOWSOUTHWEST,
-    BELOWWEST,
-    BELOWNORTHWEST,
-    CENTER,
+inline constexpr unsigned make_xyz(int const x, int const y, int const z) noexcept
```

Comment:
As it's not at all obvious at a casual glance, indeed it should have had a comment -- I'll add one.